### PR TITLE
Np 1979 transform resource to publication

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 1]
```diff
@@ -186,6 +179,22 @@ public Resource markPublicationForDeletion(Resource resource)
...
 
+    private static List<Resource> queryResultToResourceList(QueryResult result) {
+        return result.getItems()
+            .stream()
+            .map(resultValuesMap -> parseAttributeValuesMap(resultValuesMap, ResourceDao.class))
+            .map(ResourceDao::getResource)
+            .collect(Collectors.toList());
+    }
+
+    private Resource fetchEventuallyConsistentResource(Resource newResource) {
```
Comment (brinxmat) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:190]: Hmm. In my mind, this action seems like something we shouldn't be doing as the response should be 202 + location header

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java [hunk 2]
```diff
@@ -592,8 +598,20 @@ private void assertThatUpdateFails(Resource resourceUpdate) {
...
+        assertThatNewEntityDescriptionDiffersOnlyInTitle(resource.getEntityDescription(), newEntityDescription);
+        return resource.copy().withEntityDescription(newEntityDescription).build();
+
+
+    }
+
+    private void assertThatNewEntityDescriptionDiffersOnlyInTitle(EntityDescription oldEntityDescription,
+                                                                  EntityDescription newEntityDescription) {
+        Diff diff = javers.compare(oldEntityDescription, newEntityDescription);
+        assertThat(diff.getChanges().size(),is(equalTo(1)));
```
Comment (brinxmat) [inline publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:614]: You probably want to assert that the diff is the title

File: storage-model/src/test/java/no/unit/nva/publication/storage/model/ResourceTest.java [hunk 3]
```diff
@@ -40,27 +84,190 @@ public void copyContainsAllFields() {
...
+            .withDescription(randomString())
+            .withAlternativeTitles(alternativeTitles)
+            .withContributors(sampleContributor())
+            .withLanguage(SAMPLE_LANGUAGE)
+            .withMainTitle(randomString())
+            .withMetadataSource(randomUri())
+            .withNpiSubjectHeading(randomString())
+            .withTags(List.of(randomString()))
+            .withReference(reference)
+
```
Comment (brinxmat) [inline storage-model/src/test/java/no/unit/nva/publication/storage/model/ResourceTest.java:170]: Remove space

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java (hunk 1 above)
Comment (axthosarouris) [reply publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:190]: I agree, my intention was to not break the current contract. Maybe we can create a bug issue and fix it later?

**Review by brinxmat — APPROVED** [review]

**Pull request merged**